docs/data_structures/history_array.rst

```diff
@@ -7,25 +7,76 @@ history array
-must be output from ``gen_f`` or ``sim_f``. In addition to output fields, the
-final history array returned by libEnsemble will include the following
-fields, almost all of which are protected from the user. These fields are
-populated internally:
+store information about each point (ensemble member) processed in the ensemble.
+Each entry contains multiple named fields, which can contain different data types,
+including scalar and arrays.
+
+The manager maintains the global history array for the ensemble. Each entry stores
+data generated by the ``gen_f`` and then the resultant output from the ``sim_f``,
+in addition to some reserved fields.
+
+When the history array is initialized, it creates fields in ``H`` for each entry of
+``gen_specs['out']`` and ``sim_specs['out']``, provided by the user. For a simple
+example these fields might be::
+
+    gen_specs['out'] = [('x', float, 2), ('theta', int)]
+    sim_specs['out'] = [('f', float)]
+
+In the example, ``x`` is a 2 dimensional co-ordinate, ``theta`` represents some
+integer input parameter, and ``f`` is a scalar output of the simulation.
+
+For ease, the ``gen_f`` and ``sim_f`` are expected to return output as NumPy
+structured arrays. These can be created using the gen/sim_specs['out'], or a
+sub-set of these values (the manager's history array will update any fields given
+back to it.
+
+The names of the input fields for ``gen_f`` and ``sim_f`` must exist in the manager's
+history array (i.e.~ they must be output fields from ``gen_f`` or ``sim_f`` or
+reserved fields). The input fields can simply be listed, for example::
+
+    sim_specs['in'] = ['x', 'theta']
+
+
+Reserved fields in history array
+--------------------------------
+
+The manager's history array also contains a number of reserved fields. These
+include a ``sim_id`` to globally identify the point (on the manager this is
+usually the same as the array index). The ``sim_id`` can be provided by the
+user from the ``gen_f``, but is otherwise assigned by the manager as generated
+points are received.
+
+The reserved boolean field ``cancel_requested`` can also be set in a user
+function to request that libEnsemble cancels evaluation of the point.
+
+The remaining reserved fields are protected (populated by libEnsemble), and
+store information about each entry. These include boolean fields for the current
+scheduling status of the point (``given`` out for evaluation, ``returned`` from
+evaluation, and ``given_back`` to the generator). Timing fields give the time
+(since the epoch) corresponding to each state, and when the point was generated.
+Other protected fields include the worker IDs on which points were generated or
+evaluated.
+
+The user fields and the reserved fields together make up the final history array
+returned by libEnsemble.
+
+The full list of these reserved fields is given below.
+
+.. TODO separate list by protected field or field a user can set]
```

Comment:
In this PR?

Reply:
I have done this - what do you think.

Reply:
I think the break-down is right, but I wouldn't say that `sim_id` is "modifiable", because it can either be set by the user or by libE. (But it must also be in numerical order, so it's quite constrained.)

Reply:
I wasn't sure what word to use here.

Reply:
I've removed so only 'Proected fields' sub-heading.